ui/src/components/WorkerManagerWorkerTypesTable/index.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react';
+import { arrayOf, string } from 'prop-types';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import Typography from '@material-ui/core/Typography/';
+import LinkIcon from 'mdi-react/LinkIcon';
+import memoize from 'fast-memoize';
+import { camelCase } from 'change-case';
+import { isEmpty, map, pipe, sort as rSort } from 'ramda';
+import { WorkerManagerWorkerTypeSummary } from '../../utils/prop-types';
+import DataTable from '../DataTable';
+import sort from '../../utils/sort';
+import Link from '../../utils/Link';
+import DateDistance from '../DateDistance';
+
+export default class WorkerManagerWorkerTypesTable extends Component {
+  static propTypes = {
+    workerTypes: arrayOf(WorkerManagerWorkerTypeSummary).isRequired,
+    searchTerm: string,
+    path: string,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  sortWorkerTypes = memoize(
+    (workerTypes, sortBy, sortDirection, searchTerm) => {
+      const sortByProperty = camelCase(sortBy);
+      const filteredWorkerTypes = searchTerm
+        ? workerTypes.filter(({ workerType }) =>
+            workerType.includes(searchTerm)
+          )
+        : workerTypes;
+
+      return isEmpty(filteredWorkerTypes)
+        ? filteredWorkerTypes
+        : [...filteredWorkerTypes].sort((a, b) => {
+            const firstElement =
+              sortDirection === 'desc' ? b[sortByProperty] : a[sortByProperty];
+            const secondElement =
+              sortDirection === 'desc' ? a[sortByProperty] : b[sortByProperty];
+
+            return sort(firstElement, secondElement);
+          });
+    },
+    {
+      serializer: ([workerTypes, sortBy, sortDirection, searchTerm]) => {
+        const ids = pipe(
+          rSort((a, b) => sort(a.workerType, b.workerType)),
+          map(({ workerType }) => workerType)
+        )(workerTypes);
+
+        return `${ids.join('-')}-${sortBy}-${sortDirection}-${searchTerm}`;
+      },
+    }
+  );
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  renderRow = workerType => {
+    const { path } = this.props;
+    const iconSize = 16;
+    const { failedNumber, exceptionNumber, unscheduledNumber } = workerType;
+
+    return (
+      <TableRow key={workerType.name}>
+        <TableCell
+          button
+          component={Link}
+          to={`${path}/worker-types/${workerType.name}`}>
+          <Typography>{workerType.name}</Typography>
+          <LinkIcon size={iconSize} />
+        </TableCell>
```

Comment:
Let's strive to maintain consistency across the app. For links, make the link icon on the same line as the text and add the hover effect similar to https://taskcluster-web.netlify.com/provisioners/aws-provisioner-v1. You could probably just copy paste the code from another view.

![Screen Shot 2019-04-22 at 9 06 03 AM](https://user-images.githubusercontent.com/3766511/56501750-e55ae480-64dd-11e9-8553-efd199f00d8d.png)


![Screen Shot 2019-04-22 at 9 01 45 AM](https://user-images.githubusercontent.com/3766511/56501684-a462d000-64dd-11e9-80f5-98cac319990e.png)

Reply:
er, it _was_ on the same line last time I saw it

Reply:
Could you post a screenshot of how it looks on your screen, please?

Reply:
You know what - that particular code wasn't copy-pasted actually. I had questions about it. So I see we use `TableCellListItem` inside `TableCell` - why is that? There's no list that I see. Also, there's `ListItemText` that has `disableTypography` and `Typography` component inside.... the docs say that trick is used to render an alternative `Typography` component - is that what we do here?

Reply:
> There's no list in that cell. 

You're right. We used a list because we wanted to inherit the material-ui list styles. A possible fix would probably be to rename `TableCellListItem` to something else and do:

```diff
diff --git a/ui/src/components/TableCellListItem/index.jsx b/ui/src/components/TableCellListItem/index.jsx
index 45a4cae94..48f869040 100644
--- a/ui/src/components/TableCellListItem/index.jsx
+++ b/ui/src/components/TableCellListItem/index.jsx
@@ -44,9 +44,11 @@ export default class TableCellListItem extends Component {
     return (
       <List
         dense={dense}
+        component="div"
         classes={{ root: classes.root }}
         className={className}>
         <ListItem
+          component="span"
           classes={{ gutters: classes.listItem }}
           className={classes.listItemButton}
           {...props}>
```

I'll file an issue.